### PR TITLE
Clean shutdown of serviceaccount integration tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig cloud-provider

### CODE REVIEW

**Comment by aojea** [comment]
verify failure is legit

**Comment by wojtek-t** [comment]
@aojea - fixed, PTAL

File: test/integration/serviceaccount/service_account_test.go [hunk 1]
```diff
@@ -321,101 +319,104 @@ func TestServiceAccountTokenAuthentication(t *testing.T) {
...
+	// Set up a API server
+	_, clientConfig, tearDownFn := framework.StartTestServer(t, framework.TestServerSetup{
+		ModifyServerConfig: func(config *controlplane.Config) {
+			rootConfig := restclient.CopyConfig(config.GenericConfig.LoopbackClientConfig)
+			rootConfig.BearerToken = rootToken
+			rootClientset = clientset.NewForConfigOrDie(rootConfig)
+			externalRootClientset := clientset.NewForConfigOrDie(rootConfig)
+
+			externalInformers = clientinformers.NewSharedInformerFactory(externalRootClientset, controller.NoResyncPeriodFunc())
+			informers = clientinformers.NewSharedInformerFactory(rootClientset, controller.NoResyncPeriodFunc())
```
Comment (aojea) [inline test/integration/serviceaccount/service_account_test.go:344]: I see this pattern in other tests, what is the `external` suffix here meaning?

Comment (wojtek-t) [reply test/integration/serviceaccount/service_account_test.go:344]: TBH I'm not really sure those are needed.
I think that for separate clients the reason might have been separate QPS limit, but I don't think we need separate informers.
But I wanted to make those migration PRs as no-op as possible (hence decided not to touch many places that I believe can be further simplified).

**Pull request merged**